packages/relayer-lib/src/tx_builder/eth_to_cosmos.rs

```diff
@@ -0,0 +1,429 @@
+//! This module defines [`TxBuilder`] which is responsible for building transactions to be sent to
+//! the Cosmos SDK chain from events received from Ethereum.
+
+use std::time::Duration;
+
+use alloy_primitives::Address;
+use alloy_provider::Provider;
+use alloy_transport::Transport;
+use anyhow::Result;
+use ethereum_apis::{beacon_api::client::BeaconApiClient, eth_api::client::EthApiClient};
+use ethereum_light_client::consensus_state::ConsensusState;
+use ethereum_light_client::header::{AccountUpdate, TrustedSyncCommittee};
+use ethereum_light_client::{client_state::ClientState, header::Header};
+use ethereum_types::consensus::bls::BlsPublicKey;
+use ethereum_types::consensus::light_client_header::LightClientUpdate;
+use ethereum_types::consensus::slot::compute_slot_at_timestamp;
+use ethereum_types::{
+    consensus::sync_committee::compute_sync_committee_period_at_slot,
+    execution::account_proof::AccountProof,
+};
+use ibc_eureka_solidity_types::ics26::router::routerInstance;
+use ibc_proto_eureka::cosmos::tx::v1beta1::TxBody;
+use ibc_proto_eureka::google::protobuf::Any;
+use ibc_proto_eureka::ibc::core::client::v1::{Height, MsgUpdateClient};
+use ibc_proto_eureka::ibc::lightclients::wasm::v1::ClientMessage;
+use ibc_proto_eureka::ibc::{
+    core::{
+        channel::v2::{Channel, QueryChannelRequest, QueryChannelResponse},
+        client::v1::{
+            QueryClientStateRequest, QueryClientStateResponse, QueryConsensusStateRequest,
+            QueryConsensusStateResponse,
+        },
+    },
+    lightclients::wasm::v1::{
+        ClientState as WasmClientState, ConsensusState as WasmConsensusState,
+    },
+};
+use prost::Message;
+use tendermint_rpc::{Client, HttpClient};
+
+use crate::utils::{cosmos, wait_for_condition};
+use crate::{
+    chain::{CosmosSdk, EthEureka},
+    events::EurekaEvent,
+};
+
+use super::r#trait::TxBuilderService;
+
+/// The `TxBuilder` produces txs to [`CosmosSdk`] based on events from [`EthEureka`].
+#[allow(dead_code)]
+pub struct TxBuilder<T: Transport + Clone, P: Provider<T> + Clone> {
```

Comment:
We could add a `mock` field to TxBuilder so that it works in the pow case too

Reply:
Ref: #192